test/spear_test.exs

```diff
@@ -923,6 +923,36 @@ defmodule SpearTest do
       assert Spear.cancel_subscription(c.conn, subscription) == :ok
     end
 
+    test "the append/3 with `raw?: false` returns :ok", c do
+      assert :ok =
+               random_events()
+               |> Stream.take(7)
+               |> Spear.append(c.conn, c.stream_name, expect: :empty, raw?: false)
+    end
+
+    test "the append/3 with `raw?: true` returns the raw result", c do
+      assert {:ok, response} =
+               random_events()
+               |> Stream.take(7)
+               |> Spear.append(c.conn, c.stream_name, expect: :empty, raw?: true)
+
+      assert {:"event_store.client.streams.AppendResp",
+              {:success,
+               {:"event_store.client.streams.AppendResp.Success", {:current_revision, 6},
+                {:position, {:"event_store.client.streams.AppendResp.Position", _p1, _p2}}}}} =
```

Comment:
You can use the record helpers to match these more cleanly than writing out the records literally:

```elixir
assert Streams.append_resp(result: {:success, ..}) = response
```

Reply:
Pushed commit to solve for:

1. `{:ok, response} when raw? == true` case of raw response
2. Cleaned up the test append case to use the `Streams.append_resp`